### PR TITLE
Upgraded and added LLLM model downloading, percentage, and model selection

### PR DESCRIPTION
**Pull request by jwafu**
Downloading and utilizing local LLLMs and also being able to switch between different models. You can use the different features and functions around the copilot models that you find inside of the model selection dialogs in any of the Pieces products - such as delete, change models, swap between them, and more.

### CODE REVIEW

File: src/CopilotStreamController.ts [hunk 1]
```diff
@@ -45,11 +47,42 @@ export default class CopilotStreamController {
...
+        relevant: {iterable: []},
+        model: CopilotStreamController.selectedModelId
       },
     };
 
-    this.handleMessages({ input, setMessage });
+    await this.handleMessages({input, setMessage});
+  }
+
+  public async checkModel() {
```
Comment (caleb-at-pieces) [inline src/CopilotStreamController.ts:58]: Dont' think this code is used anywhere.

File: src/index.ts [hunk 2]
```diff
@@ -20,18 +21,105 @@ async function sendMessage() {
     })
 }
 
-// gets our copilot stream instance, and holds our buttons.
+// gets our copilot stream instance.
 async function main() {
     CopilotStreamController.getInstance();
+    const modelProgressController = ModelProgressController.getInstance();
```
Comment (caleb-at-pieces) [inline src/index.ts:27]: feel free to abstract some of this stuff before merging

Comment (jordan-pieces) [reply src/index.ts:27]: Going to create an issue around this abstraction for later this week and open it up for someone else to work on as well. 👍

**Review by jwafu — COMMENTED** [review]
just double checking this first push into main 👍

File: src/CopilotStreamController.ts (hunk 1 above)
Comment (jordan-pieces) [reply src/CopilotStreamController.ts]: removed this

**Pull request merged**